src/components/Select/NativeSelect.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <div class="native-select" :class="classes.select">
+    <div class="native-select__input-container">
+      <select ref="input" v-model="selected" class="native-select__input" v-bind="{ disabled, required }">
+        <template v-for="(optionOrGroup, i) in options" :key="i">
+          <template v-if="isOptionGroup(optionOrGroup)">
+            <optgroup :label="optionOrGroup.label" :disabled="optionOrGroup.disabled">
+              <template v-for="(option, x) in optionOrGroup.options" :key="x">
+                <option :value="option.value" :disabled="option.disabled">{{ option.label }}</option>
+              </template>
+            </optgroup>
+          </template>
+          <template v-else>
+            <option :value="optionOrGroup.value" :disabled="optionOrGroup.disabled">{{ optionOrGroup.label }}</option>
+          </template>
+        </template>
+      </select>
+      <div class="native-select__overlay">
+        <template v-if="isSelected">
+          <OptionLabel v-bind="selectedOption">
+            <template v-slot:label="scope">
+              <slot name="selected-option-label" v-bind="scope" />
+            </template>
+            <template v-slot:icon="scope">
+              <slot name="selected-option-icon" v-bind="scope" />
+            </template>
+          </OptionLabel>
+        </template>
+        <template v-else>
+          <span class="native-select__placeholder">
+            {{ placeholder }}
+          </span>
+        </template>
+        <i class="native-select__arrow pi pi-arrow-down-s-line pi-lg" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, prop } from 'vue-class-component'
+import { Options } from './types'
+import { getOptionFromOptionsAndGroupsByValue, isOptionGroup } from './utilities'
+import { Component } from '@/utilities/vue-class-component'
+import OptionLabel from './OptionLabel.vue'
+
+class Props {
+  modelValue = prop<string | null>({ default: null })
+  options = prop<Options>({ default: () => ({}) })
+  placeholder = prop<string>({ default: 'Choose an Option' })
+  disabled = prop<boolean | string>({ default: false })
+  required = prop<boolean | string>({ default: false })
+}
+
+@Component({
+  emits: ['update:modelValue'],
+  components: {
+    OptionLabel
+  }
+})
+export default class NativeSelect extends Vue.with(Props) {
+
+  $refs!: {
+    input: HTMLSelectElement
+  }
+
+  private isOptionGroup = isOptionGroup
+
+  get selected() {
+    return this.modelValue
+  }
+
+  set selected(value: string | null) {
+    this.$emit('update:modelValue', value)
+  }
+
+  get selectedOption() {
+    return getOptionFromOptionsAndGroupsByValue(this.options, this.selected)
+  }
+
+  get isSelected() {
+    return this.selected !== null
+  }
+
+  get isRequired() {
+    return this.required || this.required === ''
```

Comment:
why support `required` as a string?

Reply:
there's a bug in boolean props. `<myComponent disabled required />` ends up with `disabled === ''` and `required === ''`. Default value is supposed to be `true`. Saw a pr was just made yesterday to fix that but that won't be published by the time this goes out.

Reply:
To clarify the bug is in vue3 specifically part of the composition api I believe.

Reply:
I think the bug you're referring to can be resolved by explicitly calling out the type of `required` and `disabled` in the component props

Reply:
oh nice, that's much better. Added the type to the prop declaration and removed the getters. Thanks!